File: dart/lib/move_adapter.dart
import 'package:lean_dart_client/classes/player.dart';
import 'package:lean_dart_client/classes/velocity.dart';
import 'package:lean_dart_client/move_interface.dart';
import 'package:tuple/tuple.dart';
import 'package:lean_dart_client/interfaces/simple_example.dart';

class MoveAdapter {
  IMove moveInstructor;

  MoveAdapter() {
    moveInstructor = SimpleExample();
  }

  Tuple2<bool, Velocity> GetMove(PlayerInfo info) =>
      moveInstructor.performMove(info);
}

File: typescript/src/moveAdapter.ts
import IMove, { MoveResult } from "./iMove";
import { PlayerInfo } from "./classes/player";
import Velocity from "./classes/velocity";
import SimpleMoveExample from "./interfaces/simpleExample";
import DQN from "./interfaces/DQN/DQN";
import PGN from "./interfaces/PGN/PGN";

export default class MoveAdapter {
    moveInstructor: IMove;

    constructor() {
        this.moveInstructor = new DQN();
    }

    getMove(info: PlayerInfo): MoveResult {
        return this.moveInstructor.performMove(info);
    }
}
